compiler/src/dotty/tools/dotc/transform/localopt/Simplify.scala

```diff
@@ -154,14 +154,10 @@ object Simplify {
    *  System members are the only static final fields that are mutable.
    *  See https://docs.oracle.com/javase/specs/jls/se8/html/jls-17.html#jls-17.5.4
    */
-  @tailrec def isEffectivelyMutable(t: Tree)(implicit ctx: Context): Boolean = t match {
+  def isEffectivelyMutable(t: Tree)(implicit ctx: Context): Boolean = t match {
     case _ if t.symbol.is(Mutable) => true
-    case s: Select => s.symbol.owner == defn.SystemModule
-    case i: Ident  =>
-      desugarIdent(i) match {
```

Comment:
I did not understand why desugaring was needed here because it doesn't change the symbol of the tree